src/hidimstat/dcrt.py

```diff
@@ -7,304 +7,396 @@
+                self.coef_X_full = np.ravel(self.clf_screening.coef_)
+            else:
+                self.coef_X_full = self.estimated_coef
+                self.screening_threshold = 100  # remove the screening process
+            # noisy estimated coefficients is set to 0.0
+            self.non_selection = np.where(
+                np.abs(self.coef_X_full)
+                <= np.percentile(
+                    np.abs(self.coef_X_full), 100 - self.screening_threshold
+                )
+            )[0]
+            # optimisation to reduce the number of elements different to zeros
+            self.coef_X_full[self.non_selection] = 0.0
+            # select the variables for the screening
+            if self.screening:
+                selection_set = np.setdiff1d(np.arange(n_features), self.non_selection)
+                if selection_set.size == 0:
+                    self.selection_features = np.array([])
+                    self.X_residual = np.array([])
+                    self.sigma2 = np.array([])
+                    self.y_residual = np.array([])
+                    self.clf_x_residual = np.array([])
+                    self.clf_y_residual = np.array([])
+                    return self
+            else:
+                self.non_selection = []
+                selection_set = np.arange(n_features)
+            # Refit the model with the estimated support set
+            if (
+                self.refit
+                and self.estimated_coef is None
+                and selection_set.size < n_features
+            ):
+                self.clf_refit = clone(self.clf_screening)
+                self.clf_refit.fit(X_[:, selection_set], y_)
+                self.coef_X_full[selection_set] = np.ravel(self.clf_refit.coef_)
+            # For distillation of X use least_square loss
+            results = Parallel(self.n_jobs, verbose=self.joblib_verbose)(
+                delayed(_lasso_distillation_residual)(
+                    X_,
+                    y_,
+                    idx,
+                    coef_full=self.coef_X_full,
+                    sigma_X=self.sigma_X,
+                    params_lasso_distillation_x=(
+                        self.params_lasso_distillation_x
+                        if self.params_lasso_distillation_x is not None
+                        else self.params_lasso_screening
+                    ),
+                    params_lasso_distillation_y=(
+                        self.params_lasso_distillation_y
+                        if self.params_lasso_distillation_y is not None
+                        else self.params_lasso_screening
+                    ),
+                    fit_y=self.fit_y,
+                    n_jobs=1,  # the function is already called in parallel
+                    random_state=self.random_state,
+                )
+                for idx in selection_set
+            )
+        elif self.statistic == "random_forest":
```

Comment:
I think that we will have to revise the use of this so-called self.statistic.

Reply:
One way of doing it is to let the user choose its own estimator and the associate loss.

Reply:
Indeed

Reply:
let's leave that for another PR, but indeed, the estimator should be provided by the user (with a good default); I personally think RF are a good default.